scRNA-seq/01-filtering_scRNA-seq.Rmd

```diff
@@ -0,0 +1,420 @@
+                        total_counts,
+                        num_genes_exp,
+                        mito_fraction) %>%
+  # add on the cell type from metadata with left_join()
+  dplyr::left_join(dplyr::select(sc_metadata, 
+                                 geo_accession, cell_type = cell.type.ch1)) %>%
+  # sort by mito_fraction for plotting 
+  dplyr::arrange(mito_fraction) 
+  
+```
+
+Now we can plot these measures all together, along with some possible cutoffs
+
+```{r qc_scatterplot}
+ggplot(qc_df, aes (x = total_counts,
+                   y = num_genes_exp, 
+                   color = mito_fraction)) +
+  geom_point(alpha = 0.5) +
+  scale_color_viridis_c() + 
+  geom_vline(xintercept = 300000, color = "red") +
+  geom_hline(yintercept = 4000, color = "red") +
+  labs(x = "Total Count",
+       y = "Number of Genes Expressed",
+       color = "Mitochondrial\nFraction") + 
+  theme_bw()
+```
+
+
+If we want to filter our data based on these measures and cutoffs we like, we can do this with `dplyr::filter` and then select the resulting columns from the matrix.
+
+```{r qc_filter, live = TRUE}
+# create a filtered_samples data frame from qc_df
+filtered_samples <- qc_df %>%
+  dplyr::filter(total_counts > 300000,
+                num_genes_exp > 4000,
+                mito_fraction < 0.1)
+# select only passing samples for sc_data_filtered
+sc_data_filtered <- sc_data[, filtered_samples$geo_accession]
+```
+
+
+## Number of samples that express a gene as a quality measure
+
+Now we have an idea of what samples we probably want to get rid of.
+But what if our data contains genes that we can't reliably measure in these cells?
+
+Let's use our `detection_mat` to add up how many samples express each of these genes so we can try to filter out unreliably measured genes.  
+
+```{r sample_exp, live = TRUE}
+# Make a num_samples_exp vector that contains the 
+# number of cells that express a particular gene
+num_samples_exp <- rowSums(detection_mat)
+```
+
+Let's make another density plot with the number of samples that express each gene:
+
+```{r sample_exp_plot}
+# Let's take a look at what this looks like:
+ggplot(mapping = aes(x = num_samples_exp)) +
+  geom_density(fill = "lightblue") +
+  geom_vline(xintercept = 200, col = "red") +
+  xlab("Number of Samples Expressing Each Gene") + 
+  theme_classic()
+```
+
+How many genes will be excluded if we draw our cutoff at 200 samples?
+
+```{r filter_effect, live = TRUE}
+sum(num_samples_exp < 200)
+```
+That's a lot! How do we feel about that?
+
+
+To filter the genes that are not expressed in many samples, we will take a somewhat simpler approach, and use base R to select the rows from our matrix (already filtered by sample) that are above the cutoff.
+
+```{r filter_genes}
+cutoff <- 200
+# filter sc_data_filtered to only genes above cutoff value
+# save as sc_data_filtered2
+sc_data_filtered2 <- sc_data_filtered[num_samples_exp >= cutoff, ]
```

Comment:
```suggestion
sc_data_filtered_hi_exp <- sc_data_filtered[num_samples_exp >= cutoff, ]
```
Can we name it something else? I feel like participants will take this as license to start naming things tmp1, tmp2, tmp3.

Reply:
You are of course correct here, but I am going to choose a different name. `sc_data_filtered_gene_express`, I think.